### PR TITLE
comit changes necessary for imu and pos estimator unit tests

### PR DESCRIPTION
**Pull request by 2shaar2059**
## Description

necessary for imu and pose estimator unit tests. Added extra counters to relay status of what and how much data has flowed through the apps. also fixed a struct that should've been a union.
## 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have read and followed the `CONTRIBUTING.md` file
- [ ] I have applied the .clang-format file to my changes
- [ ] I have commented my code using Doxygen style comments (see Contributing MD for examples), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design documentation 
- [ ] I have updated any relevant parameters in the internal ICD
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this is a bug-fix, I have added tests that can catch similar bugs in the future
- [x] New and existing unit tests pass locally with my changes
- [ ] For cfe apps I have reserved an empty address for my msgs_ids and perf_ids from the [tracking list](https://docs.google.com/spreadsheets/d/1yRUhNTRBOiGLswlsWftNaYlwGKDeNQ_fCnDYFrc_7fU/edit#gid=772812381) and checked that it does not clash with other apps
- [ ] For new added message types, there is a timestamp included via either an explicit timestamp variable in the struct or buried in the a tlmHeader.

### CODE REVIEW

File: fsw/public_inc/imu_driver_msgs.h [hunk 1]
```diff
@@ -40,9 +40,17 @@ extern "C" {
 typedef struct {
     uint8 CommandCounter;
```
Comment (tenzincrouch) [inline fsw/public_inc/imu_driver_msgs.h:41]: @2shaar2059 - why not make these both larger, we are padding anyway? Maybe 2x int32.

**Review by tenzincrouch — DISMISSED** [review]
LGTM, please merge after addressing comment.

File: fsw/public_inc/imu_driver_msgs.h (hunk 1 above)
Comment (2shaar2059) [reply fsw/public_inc/imu_driver_msgs.h:41]: Good catch, I will enlarge and remove padding

Comment (2shaar2059) [reply fsw/public_inc/imu_driver_msgs.h:41]: actually, should we increase all similar counters to uint64?

**Review by tenzincrouch — APPROVED** [review]
LGTM.

**Pull request merged**